src/test/java/com/rackspacecloud/metrics/ingestionservice/CreateInfluxDBTests.java

```diff
@@ -0,0 +1,224 @@
+    private String openurl = "http://localhost:8086";
+    private String database = "db_1";
+    private String retentionPolicy = "autogen";
+
+    @Mock
+    private InfluxDB influxDB;
+    private NotcreatedDBInfluxDB notcreatedDBInfluxDB = new NotcreatedDBInfluxDB();
+
+    @Before
+    public void setUp() {
+        //    InfluxDB influxDB = null;
+        OkHttpClient.Builder client = new OkHttpClient.Builder()
+                .connectTimeout(1, TimeUnit.MINUTES)
+                .readTimeout(1, TimeUnit.MINUTES)
+                .writeTimeout(1, TimeUnit.MINUTES)
+                .retryOnConnectionFailure(true);
+
+        if (influxDB == null) {
+            //influxDB = InfluxDBFactory.connect("http://localhost:8086","Ravi5626","*****", client);
+            influxDB = InfluxDBFactory.connect(openurl,username,password, client);
+        }
+    }
+
+    @Test
+    public void isDBExistinRestore(){
+        //     CreateInfluxDB influxDBConnection1 = new CreateInfluxDB("","","","","");
+        CreateInfluxDB influxDBConnection1 = new CreateInfluxDB(username,password,openurl,database,retentionPolicy);
+        Query dbquery = new Query("SHOW DATABASES", null);
+        List<QueryResult.Result> dbresults = new ArrayList<>();
+        QueryResult queryRes= new QueryResult();
+        List resultList= new ArrayList();
+        resultList.add(("db_1"));
+        queryRes.setResults(resultList);
+        QueryResult result = new QueryResult();
+        result.setResults(resultList);
+        Mockito.when(influxDB.query(Mockito.any())).thenReturn(result);
+        assertTrue(result.getResults().size() == 1);
+        System.out.println("Restore database created and available=====" + resultList);
+        System.out.println("Restore database has been created");
+    }
+    @Test
+    public void isNotDBExistinRestore(){
+
+        CreateInfluxDB influxDBConnection1 = new CreateInfluxDB(username,password,openurl,database,retentionPolicy);
+        Query dbquery = new Query("SHOW DATABASES", null);
+        List<QueryResult.Result> dbresults = new ArrayList<>();
+        QueryResult queryRes= new QueryResult();
+        List resultList= new ArrayList();
+        queryRes.setResults(resultList);
+        QueryResult result = new QueryResult();
+        result.setResults(resultList);
+        Mockito.when(influxDB.query(Mockito.any())).thenReturn(result);
+        assertTrue(result.getResults().size() == 0);
+        System.out.println("Restore database not available=====" + resultList);
+
+    }
+
+    @Test
+    public void BackupDBCount(){
+        InfluxDB influxDB = notcreatedDBInfluxDB.influxDbBuild();
+        Query dbquery = new Query("SHOW DATABASES", null);
+        QueryResult queryResult = influxDB.query(dbquery);
+        List<QueryResult.Result> bkdbresults = queryResult.getResults();
+        List<List<Object>> bkfinaldbresults = bkdbresults.get(0).getSeries().get(0).getValues();
+        assertTrue(bkfinaldbresults.size() == 5);
```

Comment:
Where does "5" come from? I don't see any canned test data as part of this PR, so I'm not sure how the unit test can make such a specific assumption.